x-pack/plugins/cases/server/authorization/authorization.ts

```diff
@@ -21,25 +21,23 @@ export class Authorization {
   private readonly request: KibanaRequest;
   private readonly securityAuth: SecurityPluginStart['authz'] | undefined;
   private readonly featureCaseOwners: Set<string>;
-  private readonly isAuthEnabled: boolean;
-  // TODO: create this
-  // private readonly auditLogger: AuthorizationAuditLogger;
+  private readonly auditLogger: AuthorizationAuditLogger;
 
   private constructor({
     request,
     securityAuth,
     caseOwners,
-    isAuthEnabled,
+    auditLogger,
   }: {
     request: KibanaRequest;
     securityAuth?: SecurityPluginStart['authz'];
     caseOwners: Set<string>;
-    isAuthEnabled: boolean;
+    auditLogger: AuthorizationAuditLogger;
```

Comment:
I think this should be `PublicMethodsOf<AuthorizationAuditLogger>`

Reply:
Do we need to create a full mock for the `AuthorizationAuditLogger`? It's pretty easy to just initialize it with a fake logger since that's all it needs. We could just pass around the actual logger in our tests if we need to.

Reply:
I see what you mean. I have a dilemma about that. From the perspective of a consumer of the `AuthorizationAuditLogger` only public methods should be available and `PublicMethodsOf` achieves that. On the other hand, declaring `PublicMethodsOf` all classes that are being passed around doesn't make much sense. Maybe when we export a class (like the cases client) it is ok to use the `PublicMethodsOf` and when we use the class internally it fine to declare them without it. I am fine as it is btw 😄